internal/db/p2p_replicator.go

```diff
@@ -156,6 +157,8 @@ func (db *db) getDocsHeads(
 		}
 		defer txn.Discard(ctx)
 		ctx = SetContextTxn(ctx, txn)
+		// This is a node specific action which means the actor is the node itself.
+		ctx = identity.WithContext(ctx, db.nodeIdentity)
```

Comment:
nitpick: I don't think this function can actually guarantee that this comment is actually true, it would be better to do this in `SetReplicator` immediately after creating the ctx.  Or remove teh ctx param and create the `context.Background()` here.

Reply:
It currently is guaranteed as only `SetReplicator` calls it and the action of getting the doc heads here is used only for the purpose of a node specific action.

Reply:
moved it to `SetReplicator`.